src/components/ranking/RankingSection.tsx

```diff
@@ -98,15 +145,23 @@ const RankingSection = () => {
         onChange={(value) => handleParamChange('rank', value)}
       />
 
-      <ProductGrid
-        products={rankingProducts}
-        showMore={showMore}
-        onProductClick={handleProductClick}
-      />
+      {isLoading ? (
+        renderLoadingSkeleton()
+      ) : error ? null : ( // 에러시 아무것도 표시하지 않음
+        <>
+          <ProductGrid
+            products={data?.data || []}
+            showMore={showMore}
+            onProductClick={handleProductClick}
+          />
 
-      <MoreButton onClick={() => setShowMore(!showMore)}>
-        {showMore ? '접기' : '더보기'}
-      </MoreButton>
+          {data?.data && data.data.length > 6 && (
```

Comment:
[수정요청]

api 응답값에 따라 showMore가 true인데도 data.length가 6보다 아래인 경우가 있을 수 있습니다. 이부분은 기존처럼 showMore로 분기처리 하는 것으로 수정하는게 적절해보입니다!
```suggestion
          {data?.data && showMore && (
```

Reply:
showMore의 초기값이 false(더보기)이고, 누를때마다 true(접기)와 false(더보기)가 바뀌는 상태입니다. 제안해주신 방법으로 수정하니 초기값이 false여서 버튼이 아예 보이지 않게 되는데, 어디가 문제이고 어떻게 수정해야할지 조금 더 자세히 설명해주실 수 있나요?

Reply:
이부분은 제가 헷갈렸네요 말씀하신 대로 기존 로직이 맞네요 죄송합니다 ㅠ